admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -961,6 +961,8 @@ var form = (function() {
   }
 
   function switchLanguage(iso_code) {
+    $('#form_switch_language option[value!="' + iso_code + '"]').attr('selected', false);
+    $('#form_switch_language option[value="' + iso_code + '"]').attr('selected', true);
```

Comment:
it doesn't work... because it's a select2.. 
I got myself also

![select](https://user-images.githubusercontent.com/9074610/29418017-78df393c-836b-11e7-9428-1577e87afab1.png)